src/components/admin/AddCourse.vue

```diff
@@ -80,58 +77,137 @@ export default defineComponent({
 		const courses = useCourseStore();
 		const users = useUserStore();
 
-		const course = ref<Course>(new Course());
-		const user = ref<User>(new User());
+		const course = ref<Course>(new Course({ course_name: 'New Course' }));
+		const user = ref<User>(new User({ username: 'new_professor' }));
+
+		// modify username without editing the User
 		const username = ref<string>('');
 		const instructor_exists = ref(false);
 		const course_dates = ref<DateRange>({ to: '', from: '' });
 
+		logger.debug('[AddCourse] dialog opened.');
+
+		watch(() => course_dates, () => {
+			const start = date.extractDate(course_dates.value.from, 'YYYY/MM/DD').getTime() / 1000;
+			const end = date.extractDate(course_dates.value.to, 'YYYY/MM/DD').getTime() / 1000;
+			course.value.setDates({ start, end });
+			logger.debug(`[AddCourse/updateDates] start: ${start} end: ${end}`);
+		}, { deep: true });
+
+		const checkCourse = () => {
+			logger.debug('[AddCourse/checkCourse] The course name changed, checking for uniqueness.');
+			if (courses.getCourseByName(course.value.course_name)) {
+				logger.debug(`[AddCourse/checkCourse] A course named ${course.value.course_name} already exists.`);
+				$q.notify({
+					message: `A course named ${course.value.course_name} already exists.`,
+					color: 'red'
+				});
+			}
+		};
+
+		const addCourse = async () => {
+			let course_instructor: User | undefined;
+
+			// add global user for instructor if they don't exist yet
+			if (!instructor_exists.value) {
+				// unclear why user.value must be re-cast...
+				course_instructor = await users.addUser(user.value as User)
+					.then((_new_user) => {
```

Comment:
Shouldn't this be either await or using .then, but not both.  I'm trying to track an error down related to the course dates, which aren't updating.

Reply:
Promises can be chained regardless of await/void. In this case, we are waiting for the return value of a User ... all that's happening with `.then` is we are peeking into the returned User for the purpose of notifying through the UI.

Specifically, since we are assigning the result of this promise to a variable, we must make sure that our promise chain returns the user if the request was successful - and undef otherwise.